src/app/view/perspectives/gantt-chart/gantt-chart-tasks/gantt-chart-visualization/gantt-chart-visualization.component.ts

```diff
@@ -0,0 +1,197 @@
+
+  @Input()
+  public tasks: GanttChartTask[];
+
+  @Input()
+  public canManageConfig: boolean;
+
+  @Output()
+  public patchData = new EventEmitter<{documentId: string; attributeId: string; value: any}>();
+
+  public ganttChart: frappeGantt;
+
+  constructor(private renderer: Renderer2) {}
+
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.tasks && this.tasks) {
+      if (ganttTasksChanged(this.tasks, this.ganttChart && this.ganttChart.tasks)) {
+        this.visualize();
+      }
+    } else if (changes.config && this.config && this.ganttChart && !this.ganttChart.view_is(this.config.mode)) {
+      this.ganttChart.change_view_mode(this.config.mode);
+      this.setChartColorsAndListeners();
+    }
+  }
+
+  private visualize() {
+    if (this.ganttChart) {
+      this.refreshChart();
+    } else {
+      this.createChart();
+    }
+  }
+
+  private refreshChart() {
+    if (this.tasks.length) {
+      this.ganttChart.refresh(this.tasks);
+      this.ganttChart.change_view_mode(this.config.mode);
+      this.setChartColorsAndListeners();
+    } else {
+      this.ganttChart.clear();
+      this.ganttChart = null;
+    }
+  }
+
+  private createChart() {
+    if (this.tasks.length) {
+      this.createChartAndInitListeners(this.tasks);
+      this.setChartColorsAndListeners();
+    }
+  }
+
+  private setChartColorsAndListeners() {
+    const tasksMap = this.ganttChart.tasks.reduce((map, task) => ({...map, [task.id]: task}), {});
+
+    const disabledTaskIds = new Set();
+    const barWrappers = document.querySelectorAll('.gantt .bar-wrapper');
+    barWrappers.forEach(element => {
+      const taskId = element.getAttribute('data-id');
+      const task = tasksMap[taskId];
+      if (!task) {
+        return;
+      }
+
+      const barColor = shadeColor(task.color, 0.5);
+      const progressColor = shadeColor(task.color, 0.3);
+
+      const barElement = element.querySelector('.bar');
+      barElement && this.renderer.setStyle(barElement, 'fill', barColor);
+
+      const progressElement = element.querySelector('.bar-progress');
+      progressElement && this.renderer.setStyle(progressElement, 'fill', progressColor);
+
+      if (task.disabled) {
+        this.removeListeners(element);
+        disabledTaskIds.add(taskId);
+      }
+    });
+
+    const arrows = document.querySelector('.gantt .arrow');
+    for (let i = 0; i < arrows.children.length; i++) {
+      const arrow = arrows.children.item(i);
+      const idFrom = arrow.getAttribute('data-from');
+      const idTo = arrow.getAttribute('data-to');
+      if (disabledTaskIds.has(idFrom) || disabledTaskIds.has(idTo)) {
+        this.removeListeners(arrow);
+      }
+    }
+  }
+
+  private removeListeners(element: Element) {
+    element && element.parentNode.replaceChild(element.cloneNode(true), element);
+  }
+
+  private createChartAndInitListeners(tasks: GanttChartTask[]) {
+    this.ganttChart = new frappeGantt.default('#ganttChart', tasks, {
```

Comment:
What about appending `id` from `GanttChart` entity here? I can see that the store is prepared for multiple gantt charts shown at the same time but rendering component is not.